Assets/NavMeshComponents/Scripts/NavMeshSurface.cs

```diff
@@ -341,10 +341,11 @@ List<NavMeshBuildSource> CollectSources()
             }
 
             if (m_IgnoreNavMeshAgent)
-                sources.RemoveAll((x) => (x.component != null && x.component.gameObject.GetComponent<NavMeshAgent>() != null));
+				sources.RemoveAll((x) => x.component != null && x.component is NavMeshAgent);
```

Comment:
It may be nice but still returning true if the game object has been destroyed
Creating undesired exceptions

In this case it may be changed because it will not be q problem here, thanks for the information